coding-intro/coding-intro-core/core-concepts/organizing-commands.md

```diff
@@ -1,30 +1,33 @@
 ---
-author: kapnobatai136
+author: lior-bd
+
+tags:
+  - discussion
 
 type: normal
 
 category: must-know
-
 ---
 
-# Functions
+# Functions are powerful
 
 ---
 ## Content
 
-Functions allow us to make little snippets of code, each for a separate task that we do within our program. For instance, we are simulating a human being that loves coffee, so much that he makes 10 cups a day. For every cup of coffee he would need to do the following:
-
-1. find a clean cup
-2. boil water
-3. open the packet
-4. mix the coffee with the boiling water.
-5. drink (gulp, gulp)
+Functions are one of the most powerful tools that a programmer has.
 
-That's so much work! Instead, imagine if we had one function `make_coffee` with all of these instructions. Every time we want coffee, we write `make_coffee` and, voila! - the work is done.
+Imagine we're coding a robot chef 🤖. Making a function for turning on the oven can make our life much easier for several reasons:
 
-### Easier to read
+- **Reusability**
+  Instead of rewriting all the oven instructions 50 times, we can write one function and just call that function whenever we need
 
-Functions give the program structure and make them easy to understand. `make_bed`, `brush_teeth`, and `read_book` would be a lot easier to understand rather than a series of steps laid out one after the other.
-> 💡 You should give your functions explanatory names to make it easier for the reader to understand what they do.
+- **Take Input**
+  Sometimes we want the oven at 350 degrees, and sometimes we want it at 400. We can have our oven function take in a temperature variable as input.
 
-Can you think of 3 other functions straight out of your everyday life?
+- **Readability**
+Functions, with helpful names, can make reading our code much easier. Take for instance:
```

Comment:
This sentence more clearly described the benefits of good names:

```
You should give your functions explanatory names to make it easier for the reader to understand what they do.
````

Over this one:

```
Functions, with helpful names, can make reading our code much easier. Take for instance:
```

It's not clear what 'helpful" means and why reading the code is easier (i.e. because a helpful name describes the result achieved from a group of steps)